src/unity/toolkits/coreml_export/neural_net_models_exporter.cpp

```diff
@@ -9,31 +9,86 @@
 #include <logger/assertions.hpp>
 #include <unity/toolkits/coreml_export/mlmodel_include.hpp>
 
-#include <unity/toolkits/coreml_export/mldata_exporter.hpp>
-#include <unity/toolkits/coreml_export/coreml_export_utils.hpp>
 
 using CoreML::Specification::ArrayFeatureType;
-using CoreML::Specification::DoubleFeatureType;
 using CoreML::Specification::FeatureDescription;
 using CoreML::Specification::ImageFeatureType;
 using CoreML::Specification::ModelDescription;
 using CoreML::Specification::NeuralNetworkLayer;
 using turi::coreml::MLModelWrapper;
 
+constexpr char CONFIDENCE_STR[] = "Boxes × Class confidence (see user-defined metadata \"classes\")";
+constexpr char COORDINATES_STR[] = "Boxes × [x, y, width, height] (relative to image size)";
+constexpr char IOU_THRESHOLD_STR[] = "(optional) IOU Threshold override (default: 0.45)";
+constexpr char CONFIDENCE_THRESHOLD_STR[] = "(optional) Confidence Threshold override (default: 0.25)";
   
 namespace turi {
 
+  namespace {
+    void set_predictions_feature(FeatureDescription* feature_desc, std::string feature_name, 
+        size_t num_predictions, size_t num_classes, bool include_shape, bool include_bounds,
+        size_t bound, std::string short_desc) {
```

Comment:
When `bound` is used, isn't it always equal to `num_classes`?

For example, when num_predictions = 2535 and num_classes = 6, then we're describing a multi-array feature where the multi-array is described as either having the shape [2535, 6] or [x, 6] where x is unbounded.

Maybe `include_bounds` should just be something like `use_flexible_shape`

Reply:
Yes you're right, in the case of 'confidence' it is num_classes and in the case of 'coordinates' it is the number of coordinates i.e. 4. Based on this observation I've simplified the code by removing bounds and using number of classes or number of coordinates through num_classes. I've also changed include_bounds to use_flexible_shape.